### PR TITLE
refactor: export in index.ts (breaking change)

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

Tidy up index.ts export, before some should not be exported accidentally exported through re-export, if you use these unintentionally exported export, you may be using Rspack through the incorrect way, please stop using these hack, if there is a real need to use these exports, please raise an issue in the Rspack repository, we will evaluate it and make appropriate adjustments.

## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by h-a-n-a** [comment]
@ahabhgk  Would you fix this?

**Comment by ahabhgk** [comment]
Will rebase this before 0.4

File: crates/rspack_core/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ itertools = { workspace = true }
 mime_guess = { workspace = true }
 nodejs-resolver = { version = "0.1.0" }
 once_cell = { workspace = true }
-oxc_resolver = { version = "0.5.3", features = ["tracing-subscriber"] }
```
Comment (ahabhgk) [inline crates/rspack_core/Cargo.toml:26]: This will break RSPACK_PROFILE since there are two tracing subscriber in the global, cc @Boshen

Comment (ahabhgk) [reply crates/rspack_core/Cargo.toml:26]: Screenshot 2023-11-16 at 17 29 39

Comment (Boshen) [reply crates/rspack_core/Cargo.toml:26]: oxc_resolver has `tracing` in it, let's try and enable it on the rspack side.

**Comment by hardfist** [comment]
!eco-ci

**Comment by hardfist** [comment]
@ahabhgk please take a look the eco-ci result

**Pull request merged**